controllers/storagecluster/cephblockpools.go

```diff
@@ -69,6 +69,27 @@ func (r *StorageClusterReconciler) newCephBlockPoolInstances(initData *ocsv1.Sto
 			},
 		},
 	}
+
+	// create `.nfs` cephblockpool if NFS is enabled.
+	if initData.Spec.NFS != nil && initData.Spec.NFS.Enable {
+		ret = append(ret,
+			&cephv1.CephBlockPool{
```

Comment:
This is not a pool for block images, it is a RADOS pool that contains objects which are snippets of NFS-Ganesha configuration files. Depending on the Ceph version, the name of the pool should be "nfs-ganesha", or for more recent versions ".nfs" as you have now.

I would have expected Rook to create the pool with the right name when the CephNFS resource is initialized. But maybe there is a reason it should be done here?

Reply:
> This is not a pool for block images, it is a RADOS pool that contains objects which are snippets of NFS-Ganesha configuration files. Depending on the Ceph version, the name of the pool should be "nfs-ganesha", or for more recent versions ".nfs" as you have now.

This feature will be present from odf 4.11 onwards, which will have ceph pacific 16.2.7+ 
We don't need go worry about backward support as far as we discussed
@

> I would have expected Rook to create the pool with the right name when the CephNFS resource is initialized. But maybe there is a reason it should be done here?

So that extra parameters can be added, replication spec and failure domain 


BlaineEXE can confirm both the above statement.